doc/api/stream.md

```diff
@@ -1889,7 +1889,51 @@ const stream = addAbortSignal(
   }
 })();
 ```
-## API for stream implementers
+
+### stream.Transform.by(asyncGeneratorFunction[, options])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `asyncGeneratorFunction` {AsyncGeneratorFunction} A mapping function which
+accepts a `source` async iterable which can be used to read incoming data, while
+transformed data is pushed to the stream with the `yield` keyword.
+* `options` {Object} Options provided to `new stream.Transform([options])`.
+By default, `Transform.by()` will set `options.objectMode` to `true`,
+unless this is explicitly opted out by setting `options.objectMode` to `false`.
+* Returns: {stream.Transform}
+
+A utility method for creating Transform Streams with async generator functions.
+The async generator is supplied a single argument, `source`, which is used to
+read incoming chunks.
+
+Yielded values become the data chunks emitted from the stream.
+
+```js
```

Comment:
@aduh95 Will this still render correctly or are we in a world now where all examples need to be written twice, once for ESM and once for CommonJS?

Reply:
They don't _need_ to be written twice, although I personally think having ESM examples available improves our docs. In this case `stream.md` contains solely CJS-only code snippets, porting those to ESM should be done in another PR, it would be a bit weird to have just that one available in two flavors IMHO.